object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -150,7 +167,20 @@ boolean returnsThis(final MethodInvocationTree tree) {
     AnnotatedTypeMirror methodATm = rrATF.getAnnotatedType(methodEle);
     AnnotatedTypeMirror rrType =
         ((AnnotatedTypeMirror.AnnotatedExecutableType) methodATm).getReturnType();
-    return rrType != null && rrType.hasAnnotation(This.class);
+    if (rrType != null && rrType.hasAnnotation(This.class)) {
```

Comment:
Just write this `if` condition as an or?

Reply:
Good thought. Done.